core/domain/exp_domain.py

```diff
@@ -2499,3 +2499,470 @@ def add_contribution_by_user(self, contributor_id):
+                    change_is_mergeable = True
+                elif (change.property_name ==
+                      STATE_PROPERTY_LINKED_SKILL_ID):
+                    change_is_mergeable = True
+                # We’ll not be able to handle the merge if changelists
+                # affect the different indices of the hint in the same
+                # state because whenever there is even a small change
+                # in one field of any hint, they treat the whole hints
+                # list as a new value.
+                # So it will not be possible to find out the exact change.
+                elif (change.property_name ==
+                      STATE_PROPERTY_INTERACTION_HINTS):
+                    if state_name in self.changed_properties:
+                        if (change.property_name not in
+                                self.changed_properties[state_name] and
+                                change.property_name not in
+                                self.changed_translations[state_name]):
+                            change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+                elif (change.property_name ==
+                      STATE_PROPERTY_INTERACTION_SOLUTION):
+                    if state_name in self.changed_properties:
+                        if (old_exp_states.interaction.id ==
+                                current_exp_states.interaction.id):
+                            if (all(property not in
+                                    self.changed_properties[state_name]
+                                    for property in
+                                    (self
+                                     .PROPERTIES_CONFLICTING_SOLUTION_CHANGES
+                                    )) and
+                                    change.property_name not in
+                                    self.changed_translations[state_name] and
+                                    STATE_PROPERTY_INTERACTION_SOLUTION not in
+                                    self.changed_properties[state_name]):
+                                change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+                elif (change.property_name ==
+                      STATE_PROPERTY_SOLICIT_ANSWER_DETAILS):
+                    if state_name in self.changed_properties:
+                        if (old_exp_states.interaction.id ==
+                                current_exp_states.interaction.id and
+                                old_exp_states.solicit_answer_details ==
+                                current_exp_states.solicit_answer_details):
+                            change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+                elif (change.property_name ==
+                      STATE_PROPERTY_CARD_IS_CHECKPOINT):
+                    change_is_mergeable = True
```

Comment:
I see this pattern for multiple commands, please create NON_CONFLICTING_PROPERTIES or have change.property_name in [...., ..., ...].

Reply:
Done.